src/js/d3Utils.js

```diff
@@ -23,6 +23,37 @@ https://raw.githubusercontent.com/fluid-project/chartAuthoring/master/LICENSE.tx
         return d3.selectAll(elem.toArray());
     };
 
+    /* Given the object "d", returns value of:
+     * - "id" if it exists on the top-level object,
+     * - "data.id" if it exists on a "data" object
+     * - "undefined" if neither of these exist
+     */
+    floe.d3.idExtractor = function (d) {
+        return d.id !== undefined ? d.id : d.data !== undefined ? d.data.id !== undefined ? d.data.id : undefined : undefined;
+    };
+
+    // Given a selection of D3 elements and an ID, returns only the elements
+    // matching that ID
+    floe.d3.filterById = function (d3Selection, currentlyPlayingDataId) {
```

Comment:
```floe.d3.filterById()``` and ```floe.d3.filterByNotId``` are almost identical. Can we combine them into one function by passing in a flag parameter? If we do so, the name of the function might need to be changed to match the intention of the flag.